### ISSUE TITLE
`exwm` does not work correctly

### ISSUE THREAD

**Issue by guskovd** [emacs-ng/emacs-ng#333]
I was unable to run exwm on the latest version of emacs-ng. When I start the X server, I see a white screen

**Comment by brotzeit** [comment]
So it worked before the recent upstream merge #327 ? If so, does exwm work with latest emacs master ?

**Comment by guskovd** [comment]
I've been using the old [version](https://github.com/emacs-ng/emacs-ng/commit/cdb88483ac932cb8109fc24287bfc80e4c661b67) for a long time:

Everything worked well

After that, I tried to update emacs-ng several times but ran into this error. I think it broke much earlier than https://github.com/emacs-ng/emacs-ng/pull/327

**Comment by brotzeit** [comment]
What were the options you passed to configure ?

**Comment by guskovd** [comment]
Even the simplest config does not work for me:
```
(require 'exwm)
(require 'exwm-config)
(exwm-config-default)
```

**Comment by ykhurshid** [comment]
I'm pretty sure he's talking about your emacs-ng build configuration, not your init file.

**Comment by brotzeit** [comment]
@ykhurshid thanks for clarifying, but I already assumed he just ran `./configure` without arguments.

**Comment by guskovd** [comment]
> I'm pretty sure he's talking about your emacs-ng build configuration, not your init file.

I am building a package with `flake.nix` from the last commit

**Comment by brotzeit** [comment]
What was the last working commit ?

**Comment by guskovd** [comment]
I can tell that `exwm` worked fine in commit: https://github.com/emacs-ng/emacs-ng/commit/cdb88483ac932cb8109fc24287bfc80e4c661b67
There are a lot of changes between this commit and the last one.
`emacs-ng` takes a long enough time to figure out where it broke, and it takes too much time and effort.

**Comment by brotzeit** [comment]
Yeah I didn't expect that you determine the exact commit, I just meant what commit you were using before.

**Comment by LemonBreezes** [comment]
Hey, so I was recompiling `emacs-ng` over and over today trying to figure out why `exwm` wasn't working and I realized that `exwm` breaks when `emacs-ng` is configured with `--with-webrender`. It seems like part of the problem is that several functions `exwm` uses such as `x-display-screens` and `x-get-atom-name` are in C files which are excluded from Emacs when `webrender` is enabled. I also noticed Emacs does not function correctly in X11 without a window manager when `webrender` is enabled (for example, `toggle-frame-fullscreen` does not work). I would love to see this fixed because I am excited to try out `webrender` alongside my `exwm` workflow.
